### PR TITLE
Fix escaping of trailing quote in quoted identifiers

### PR DESCRIPTION
**Pull request by razzolini-qpq**
The current `Display` impl for `Ident`
https://github.com/sqlparser-rs/sqlparser-rs/blob/a6d7a35dacbcfe282d1cae83916505c6294edf4e/src/ast/mod.rs#L130-L141
fails to escape a quote (', " or \`) when it's the last character in the identifier's value. Specifically, this quote is only written to the string once, rather than twice. For example, `Ident::with_quote('\'', "quoted'").to_string()` returns `"'quoted''"` (with two trailing single quotes) instead of `"'quoted'''"` (with three trailing single quotes — two for the escaped quote and one for the identifier's closing quote). This is because escaping is implemented by prepending an additional quote to each of the substrings returned by `split_inclusive` (except for the first), but `split_inclusive` (unlike `split`) *doesn't* return an empty substring after a trailing separator character.

Instead of trying to fix this code directly, I've reused the `EscapeSingleQuoteString` struct, which already provided a correct implementation of escaping for single-quoted strings, and was easily generalized to work with an arbitrary quote character.

### CODE REVIEW

File: tests/sqlparser_mysql.rs [hunk 1]
```diff
@@ -293,7 +293,7 @@ fn parse_quote_identifiers() {
 
 #[test]
 fn parse_quote_identifiers_2() {
-    let sql = "SELECT `quoted `` identifier`";
+    let sql = "SELECT ```quoted `` identifier```";
     assert_eq!(
         mysql().verified_stmt(sql),
```
[the remark this answers is not in the document]
Comment (razzolini-qpq) [reply tests/sqlparser_mysql.rs]: > Can you please leave the existing test and add a new test instead `parse_quote_identifiers_3`?

I've restored the original tests, then added a `parse_quote_identifiers_3` test for MySQL and a `parse_quoted_identifier_2` test for PostgreSQL.

In my tests, should I also remove the escaped quote at the start of the identifiers? The bug only affected the quote at the end, but I think that testing both of these edge cases might be useful to catch any future regressions.

[the remark this answers is not in the document]
Comment (alamb) [reply tests/sqlparser_mysql.rs]: > In my tests, should I remove the escaped quote at the start of the identifiers, or maybe move that to a separate test, too? The bug only affected the quote at the end, but I think that testing both of these edge cases might be useful to catch any future regressions.

I am always a fan of additional test coverage, especially around corner cases. So yes please!

**Review by alamb — APPROVED** [review]
Looks great @razzolini-qpq  thank you very much. Let me know if you want to add the additional test coverage in this PR or if you plan another PR

**Comment by razzolini-qpq** [comment]
> Looks great @razzolini-qpq thank you very much. Let me know if you want to add the additional test coverage in this PR or if you plan another PR

Both corner cases — a quote at the start and one at the end of the identifier — are already covered by the tests that are currently in this PR. Specifically, the `parse_quote_identifiers_3` test covers both corner cases for the MySQL dialect, and the `parse_quoted_identifier_2` test does the same for the PostgreSQL and generic dialects. Would you prefer separate tests for each of the two corner cases?

**Comment by alamb** [comment]
>  Would you prefer separate tests for each of the two corner cases?

Nope -- this sounds great. Thank you
